safe-global/safe-wallet-web
# feat: pending recovery widget

## What it solves

Resolves #2754

## How this PR fixes it

A pending recovery widget has been added to the top of the dashboard according to [the designs](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=457-8024&mode=design&t=0u4GUy7BGLZdRDRR-0):

- When a pending recovery has not expired and is invalid, a countdown is shown.
- When a pending recovery has not expired and is valid, a warning is shown. (Note: this will be adjusted when #2758 is implemented)

Full test coverage has been included for UI elements and associated logic.

## How to test it

Open a Safe that has a pending recovery attempt, e.g. `gor:0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b` and observe the new banner.

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/56b55484-265d-4881-b22a-4c3842b547ca)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/79570fe5-d725-4baf-b8a2-a5e17af23733)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/7a2c8e22-7e23-46ab-b91e-9cbb96978964)

## Checklist
